src/bun.js/node/types.zig

```diff
@@ -2848,20 +2865,35 @@ pub const Path = struct {
         lenSum += 1;
 
         const pathObject = .{ .root = root, .dir = dir, .base = base, .ext = ext, .name = _name };
-        const bufLen = @max(lenSum, MAX_PATH_BYTES);
-        if (bufLen > MAX_PATH_BYTES) {
+        const size = PathSize(u8);
+        const bufLen = getPathLenT(u8, lenSum);
+        if (bufLen > size.MAX) {
             const buf = allocator.alloc(u8, bufLen) catch bun.outOfMemory();
             defer allocator.free(buf);
             return if (isWindows)
                 toJSString(globalObject, formatWindow(pathObject, buf))
             else
                 toJSString(globalObject, formatPosix(pathObject, buf));
         }
-        var buf: PathBuffer = undefined;
+        const buf = switch (bufLen) {
+            size.MAX => brk: {
+                var _buf: [size.MAX]u8 = undefined;
+                break :brk &_buf;
```

Comment:
i'm not sure if this is actually is an issue but i get weary of keeping a pointer stack memory in a block. you should think of the `var` above and any pointers into it as being invalid once the block containing it ends.

but this whole part is invalid for the other reason

Reply:
I don't think this is an issue but the one above it is totally a biggie.